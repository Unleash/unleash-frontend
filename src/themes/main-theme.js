import { createMuiTheme } from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: "#009688",
            light: "#B2DFDB",
            dark: "#00796B"
        },
        secondary: {
            main: "#00BCD4"
        }
    }
});

export default theme;
