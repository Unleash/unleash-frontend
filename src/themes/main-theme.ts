import { createTheme, Theme } from '@material-ui/core/styles';

type MainTheme = typeof mainTheme;

declare module '@material-ui/core/styles/createTheme' {
    interface Theme extends MainTheme {}
}

declare module '@material-ui/core/styles/makeStyles' {
    interface Theme extends MainTheme {}
}

declare module '@material-ui/core/styles/useTheme' {
    interface Theme extends MainTheme {}
}

const mainTheme = {
    typography: {
        fontFamily: ['Sen', 'Roboto, sans-serif'],
        fontWeightBold: '700',
        fontWeightMedium: '700',
    },
    palette: {
        primary: {
            main: '#635DC5',
            light: '#817AFE',
            dark: '#635DC5',
        },
        secondary: {
            main: '#635DC5',
            light: '#817AFE',
            dark: '#635DC5',
        },
        grey: {
            main: '#6C6C6C',
            light: '#7e7e7e',
        },
        neutral: {
            main: '#18243e',
        },
        icons: {
            lightGrey: '#e0e0e0',
        },
        chips: {
            main: '#b0bec5',
        },
        searchField: {
            main: '#fff',
        },
        iconButtons: {
            main: '#fff',
        },
        tabs: {
            main: '#efefef',
        },
        links: {
            deprecated: '#1d1818',
        },
        borders: {
            main: '#f1f1f1',
        },
        error: {
            main: '#d95e5e',
        },
        success: {
            main: '#3bd86e',
        },
        division: {
            main: '#f1f1f1',
        },
        footer: {
            main: '#000',
            background: '#fff',
        },
        code: {
            main: '#0b8c8f',
            diffAdd: 'green',
            diffSub: 'red',
            diffNeutral: 'black',
            edited: 'blue',
            background: '#efefef',
        },
        cards: {
            gradient: {
                top: '#265f6d',
                bottom: '#1A4049',
            },
            container: {
                bg: '#f1f1f1',
            },
        },
        login: {
            gradient: {
                top: '#607D8B',
                bottom: '#173341',
            },
            main: '#fff',
        },
        dialogue: {
            title: {
                main: '#fff',
            },
        },
    },
    padding: {
        pageContent: {
            header: '1.8rem 2rem',
            body: '2rem',
        },
    },
    borders: {
        default: '1px solid #f1f1f1',
        radius: { main: '3px' },
    },
    fontSizes: {
        mainHeader: '1.2rem',
        subHeader: '1.1rem',
        bodySize: '1rem',
        smallBody: '0.9rem',
        smallerBody: '0.8rem',
    },
    boxShadows: {
        chip: {
            main: `0 2px 2px 0 rgba(0, 0, 0, 0.14),
        0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)`,
        },
    },
    fontWeight: {
        thin: 300,
        medium: 400,
        semi: 700,
        bold: 700,
    },
};

export default createTheme(mainTheme as unknown as Theme);
