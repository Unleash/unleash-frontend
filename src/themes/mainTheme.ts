import { createTheme, Theme, adaptV4Theme } from '@mui/material/styles';

type MainTheme = typeof mainTheme;

declare module '@mui/material/styles/createTheme' {
    interface Theme extends MainTheme {}
}

declare module '@mui/material/styles/makeStyles' {
    interface Theme extends MainTheme {}
}

declare module '@mui/material/styles/useTheme' {
    interface Theme extends MainTheme {}
}

const mainTheme = {
    typography: {
        fontFamily: ['Sen', 'Roboto, sans-serif'],
        fontWeightBold: '700',
        fontWeightMedium: '700',
    },
    palette: {
        primary: {
            main: '#635DC5',
            light: '#817AFE',
            dark: '#635DC5',
        },
        secondary: {
            main: '#635DC5',
            light: '#817AFE',
            dark: '#635DC5',
        },
        grey: {
            main: '#6C6C6C',
            light: '#F6F6FA',
        },
        neutral: {
            main: '#18243e',
        },
        icons: {
            lightGrey: '#e0e0e0',
        },
        chips: {
            main: '#b0bec5',
        },
        searchField: {
            main: '#fff',
        },
        iconButtons: {
            main: '#fff',
        },
        tabs: {
            main: '#efefef',
        },
        links: {
            deprecated: '#1d1818',
        },
        borders: {
            main: '#f1f1f1',
        },
        error: {
            main: '#d95e5e',
        },
        success: {
            main: '#3bd86e',
        },
        division: {
            main: '#f1f1f1',
        },
        footer: {
            main: '#000',
            background: '#fff',
        },
        code: {
            main: '#0b8c8f',
            diffAdd: 'green',
            diffSub: 'red',
            diffNeutral: 'black',
            edited: 'blue',
            background: '#efefef',
        },
        cards: {
            gradient: {
                top: '#265f6d',
                bottom: '#1A4049',
            },
            container: {
                bg: '#f1f1f1',
            },
        },
        login: {
            gradient: {
                top: '#607D8B',
                bottom: '#173341',
            },
            main: '#fff',
        },
        dialogue: {
            title: {
                main: '#fff',
            },
        },
    },
    padding: {
        pageContent: {
            header: '1.8rem 2rem',
            body: '2rem',
        },
    },
    borders: {
        default: '1px solid #f1f1f1',
        radius: { main: '3px' },
    },
    fontSizes: {
        mainHeader: '1.2rem',
        subHeader: '1.1rem',
        bodySize: '1rem',
        smallBody: '0.9rem',
        smallerBody: '0.8rem',
    },
    boxShadows: {
        chip: {
            main: `0 2px 2px 0 rgba(0, 0, 0, 0.14),
        0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)`,
        },
    },
    fontWeight: {
        thin: 300,
        medium: 400,
        semi: 700,
        bold: 700,
    },
    v2: {
        palette: {
            primary: '',
            grey: {
                '10': '#F7F7FA',
                '20': '#F2F2F5',
                '30': '#EAEAED',
                '40': '#E1E1E3',
                '50': '#BDBDBF',
                '70': '#78787A',
                '90': '#202021',
            },
        },
        fontSizes: {
            headerIcon: '18px',
        },
        boxShadows: {
            primary: '0px 2px 4px rgba(129, 122, 254, 0.2)',
        },
    },
};

export default createTheme(adaptV4Theme(mainTheme as unknown as Theme));
