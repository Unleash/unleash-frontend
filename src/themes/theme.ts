import { createTheme } from '@mui/material/styles';
import { colors } from './colors';

export default createTheme({
    boxShadows: {
        main: '0px 2px 4px rgba(129, 122, 254, 0.2)',
    },
    typography: {
        fontFamily: 'Sen, Roboto, sans-serif',
        fontWeightBold: '700',
        fontWeightMedium: '700',
        allVariants: { lineHeight: 1.4 },
        button: { lineHeight: 1.75 },
        h1: {
            fontSize: '1.5rem',
            lineHeight: 1.875,
        },
    },
    fontSizes: {
        mainHeader: '1.25rem',
        subHeader: '1.1rem',
        bodySize: '1rem',
        smallBody: '0.9rem',
        smallerBody: '0.8rem',
    },
    fontWeight: {
        thin: 300,
        medium: 400,
        semi: 700,
        bold: 700,
    },
    shape: {
        borderRadius: '3px',
        borderRadiusLarge: '10px',
        borderRadiusExtraLarge: '25px',
    },
    palette: {
        primary: {
            main: colors.purple[800],
            light: colors.purple[700],
            dark: colors.purple[900],
        },
        secondary: {
            main: colors.purple[800],
            light: colors.purple[700],
            dark: colors.purple[900],
        },
        info: {
            light: colors.blue[700],
            main: colors.blue[700],
            dark: colors.blue[800],
        },
        success: {
            light: colors.green[700],
            main: colors.green[700],
            dark: colors.green[800],
        },
        warning: {
            light: colors.orange[700],
            main: colors.orange[700],
            dark: colors.orange[800],
        },
        error: {
            light: colors.red[700],
            main: colors.red[700],
            dark: colors.red[800],
        },
        divider: colors.grey[300],
        dividerAlternative: colors.grey[500],
        grey: colors.grey,
        text: {
            primary: colors.grey[900],
            secondary: colors.grey[800],
            disabled: colors.grey[600],
        },
        code: {
            main: '#0b8c8f',
            diffAdd: 'green',
            diffSub: 'red',
            diffNeutral: 'black',
            edited: 'blue',
            background: '#efefef',
        },
        activityIndicators: {
            unknown: colors.grey[100],
            recent: colors.green[100],
            inactive: colors.orange[200],
            abandoned: colors.red[200],
        },
    },
    components: {
        MuiTableHead: {
            styleOverrides: {
                root: {
                    background: colors.grey[200],
                },
            },
        },
        MuiLink: {
            styleOverrides: {
                root: {
                    color: colors.purple[900],
                },
            },
        },
        MuiBreadcrumbs: {
            styleOverrides: {
                root: {
                    '& a': {
                        color: colors.purple[900],
                    },
                },
            },
        },
        MuiTableRow: {
            styleOverrides: {
                root: {
                    '&.MuiTableRow-hover:hover': {
                        background: colors.grey[100],
                    },
                },
            },
        },
    },
});
