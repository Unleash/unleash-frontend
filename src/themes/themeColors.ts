export const unleashGrey = {
    '100': '#f7f7fa',
    '200': '#f2f2f5',
    '300': '#eaeaed',
    '400': '#e1e1e3',
    '500': '#bdbdbf',
    '700': '#78787a',
    '900': '#202021',
};
