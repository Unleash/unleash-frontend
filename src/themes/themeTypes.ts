declare module '@mui/material/styles' {
    interface CustomTheme {
        /**
         * @deprecated
         */
        fontSizes: {
            mainHeader: string;
            subHeader: string;
            bodySize: string;
            smallBody: string;
            smallerBody: string;
        };
        /**
         * @deprecated
         */
        fontWeight: {
            thin: number;
            medium: number;
            semi: number;
            bold: number;
        };
        /**
         * @deprecated
         */
        boxShadows: {
            main: string;
        };
    }

    interface CustomPalette {
        /**
         * Colors for event log output.
         */
        code: {
            main: string;
            diffAdd: string;
            diffSub: string;
            diffNeutral: string;
            edited: string;
            background: string;
        };
        /**
         * For 'Seen' column on feature toggles list.
         */
        activityIndicators: {
            unknown: string;
            recent: string;
            inactive: string;
            abandoned: string;
        };
        dividerAlternative: string;
        /**
         * For table header hover effect.
         */
        tableHeaderHover: string;
        /**
         * Text highlight effect color. Used when filtering/searching over content.
         */
        highlight: string;
        /**
         * For sidebar container background.
         */
        sidebarContainer: string;
    }

    interface Theme extends CustomTheme {}
    interface ThemeOptions extends CustomTheme {}

    interface Palette extends CustomPalette {}
    interface PaletteOptions extends CustomPalette {}
}

declare module '@mui/system/createTheme/shape' {
    interface Shape {
        borderRadiusMedium: string;
        borderRadiusLarge: string;
        borderRadiusExtraLarge: string;
    }
}

export {};
