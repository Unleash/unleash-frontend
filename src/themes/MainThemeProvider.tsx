import React, { FC } from 'react';
import {
    CssBaseline,
    ThemeProvider,
    Theme,
    StyledEngineProvider,
} from '@mui/material';
import StylesProvider from '@mui/styles/StylesProvider';
import mainTheme from './mainTheme';

declare module '@mui/styles/defaultTheme' {
    // eslint-disable-next-line @typescript-eslint/no-empty-interface
    interface DefaultTheme extends Theme {}
}

export const MainThemeProvider: FC = ({ children }) => (
    <StyledEngineProvider injectFirst>
        <ThemeProvider theme={mainTheme}>
            <StylesProvider injectFirst>
                <CssBaseline />
                {children}
            </StylesProvider>
        </ThemeProvider>
    </StyledEngineProvider>
);
