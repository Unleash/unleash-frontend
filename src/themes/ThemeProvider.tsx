import React, { FC } from 'react';
import {
    CssBaseline,
    ThemeProvider as MuiThemeProvider,
    StyledEngineProvider,
} from '@mui/material';
import mainTheme from 'themes/theme';

export const ThemeProvider: FC = ({ children }) => (
    <StyledEngineProvider injectFirst>
        <MuiThemeProvider theme={mainTheme}>
            <CssBaseline />
            {children}
        </MuiThemeProvider>
    </StyledEngineProvider>
);
