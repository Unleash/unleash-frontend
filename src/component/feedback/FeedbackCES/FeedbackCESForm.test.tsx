import renderer from 'react-test-renderer';
import { FeedbackCESForm } from './FeedbackCESForm';
import { ThemeProvider } from 'themes/ThemeProvider';

test('FeedbackCESForm', () => {
    const onClose = () => {
        throw new Error('Unexpected onClose call.');
    };

    const tree = renderer.create(
        <ThemeProvider>
            <FeedbackCESForm
                onClose={onClose}
                state={{ title: 'a', text: 'b', path: '/c' }}
            />
        </ThemeProvider>
    );

    expect(tree).toMatchSnapshot();
});
