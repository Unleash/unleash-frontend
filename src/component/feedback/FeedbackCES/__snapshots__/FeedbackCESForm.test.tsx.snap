// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackCESForm 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-1"
    >
      <h1
        class="makeStyles-title-3"
      >
        Please help us improve
      </h1>
      <form
        aria-live="polite"
        class="makeStyles-form-2"
      >
        <p
          class="makeStyles-subtitle-4"
        >
          a
        </p>
        <div
          class="makeStyles-scoreInput-8"
        >
          <span
            class="makeStyles-scoreHelp-9"
          >
            Very difficult
          </span>
          <label
            class="makeStyles-scoreValue-10"
          >
            <input
              name="score"
              type="radio"
              value="1"
            />
            <span>
              1
            </span>
          </label>
          <label
            class="makeStyles-scoreValue-10"
          >
            <input
              name="score"
              type="radio"
              value="2"
            />
            <span>
              2
            </span>
          </label>
          <label
            class="makeStyles-scoreValue-10"
          >
            <input
              name="score"
              type="radio"
              value="3"
            />
            <span>
              3
            </span>
          </label>
          <label
            class="makeStyles-scoreValue-10"
          >
            <input
              name="score"
              type="radio"
              value="4"
            />
            <span>
              4
            </span>
          </label>
          <label
            class="makeStyles-scoreValue-10"
          >
            <input
              name="score"
              type="radio"
              value="5"
            />
            <span>
              5
            </span>
          </label>
          <label
            class="makeStyles-scoreValue-10"
          >
            <input
              name="score"
              type="radio"
              value="6"
            />
            <span>
              6
            </span>
          </label>
          <label
            class="makeStyles-scoreValue-10"
          >
            <input
              name="score"
              type="radio"
              value="7"
            />
            <span>
              7
            </span>
          </label>
          <span
            class="makeStyles-scoreHelp-9"
          >
            Very easy
          </span>
        </div>
        <div
          hidden=""
        >
          <label
            class="makeStyles-textLabel-5"
            for="comment"
          >
            b
          </label>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-opjsp0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-1"
                rows="3"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-buttons-6"
          hidden=""
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled makeStyles-button-7 css-2lddgn-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            Send feedback
          </button>
        </div>
      </form>
    </div>
    <div
      aria-atomic="true"
      aria-live="polite"
      class="makeStyles-container-11"
      data-testid="ANNOUNCER_ELEMENT_TEST_ID"
      role="status"
    />
  </div>
</body>
`;
